packages/smooth_app/lib/helpers/analytics_helper.dart

```diff
@@ -0,0 +1,265 @@
+  data.addIfVAndNew('lang', Localizations.localeOf(context).languageCode);
+  data.addIfVAndNew('country', Localizations.localeOf(context).countryCode);
+
+  return _track(
+    _initAction,
+    data,
+  );
+}
+
+// TODO(m123): Matomo removes leading 0 from the barcode
+Future<bool> trackScannedProduct({required String barcode}) => _track(
+      _scanAction,
+      <String, String>{
+        eventCategory: 'Scanner',
+        eventAction: 'Scanned',
+        eventValue: barcode,
+      },
+    );
+
+Future<bool> trackProductPageOpen({
+  required Product product,
+}) {
+  final Map<String, String> data = <String, String>{
+    eventCategory: 'Product page',
+    eventAction: 'opened',
+  };
+  data.addIfVAndNew(eventValue, product.productName);
+  data.addIfVAndNew(eventName, product.productName);
+
+  return _track(
+    _productPageAction,
+    data,
+  );
+}
+
+// TODO(m123): Check where to call
+Future<bool> trackKnowledgePanelOpen({
+  required String barcode,
+  String? knowledgePanelName,
+}) {
+  final Map<String, String> data = <String, String>{
+    eventCategory: 'Knowledge panel',
+    eventAction: 'opened',
+    eventValue: barcode,
+  };
+  data.addIfVAndNew(eventName, knowledgePanelName);
+
+  return _track(
+    _knowledgePanelAction,
+    data,
+  );
+}
+
+Future<bool> trackPersonalizedRanking({
+  required String title,
+  required int products,
+  required int goodProducts,
+  required int badProducts,
+  required int unknownProducts,
+}) =>
+    _track(
+      _personalizedRankingAction,
+      <String, String>{
+        'title': title,
+        'productsCount': '$products',
+        'goodProducts': '$goodProducts',
+        'badProducts': '$badProducts',
+        'unkownProducts': '$unknownProducts',
+      },
+    );
+
+void trackSearch({
+  required String search,
+  String? searchCategory,
+  int? searchCount,
+}) {
+  final Map<String, String> data = <String, String>{
+    'search': search,
+  };
+  data.addIfVAndNew('search_cat', searchCategory);
+  data.addIfVAndNew('search_count', searchCount);
+
+  if (data.toString() == latestSearch) {
+    return;
+  }
+  latestSearch = data.toString();
+
+  _track(
+    _searchAction,
+    data,
+  );
+}
+
+Future<bool> trackOpenLink({required String url}) => _track(
+      _linkAction,
+      <String, String>{
+        'url': url,
+        'link': url,
+      },
+    );
+
+Future<bool> _track(String actionName, Map<String, String> data) {
+  final DateTime date = DateTime.now();
+  final Map<String, String> addedData = <String, String>{
+    'action_name': actionName,
+    'rand': Random().toString(),
+    'h': date.hour.toString(),
+    'm': date.minute.toString(),
+    's': date.second.toString(),
+  };
+  addedData.addIfVAndNew('uid', _getId());
+  addedData.addAll(data);
+
+  return MatomoForever.sendDataOrBulk(addedData);
+}
+
+String? _getId() {
+  return kDebugMode ? 'smoothie-debug' : OpenFoodAPIConfiguration.uuid;
+}
+
+class TrackingHelper {
```

Comment:
Given that this class only deals with `LocalDatabase`, perhaps `TrackingDatabaseHelper`?